### PR TITLE
ETD-196 Update transfer submission UI

### PR DESCRIPTION
**Pull request by jazairi**
#### Why these changes are being introduced:

The UX review of the transfer submission UI recommended some changes.

#### Relevant ticket(s):

* https://mitlibraries.atlassian.net/browse/ETD-196
* https://mitlibraries.atlassian.net/browse/ETD-80 (review and mockup)

#### How this addresses that need:

This updates the transfer submission UI to reflect the mockup in ETD-80.

#### Side effects of this change:

Adds an attr_accessor method to the Transfer model to make the
certification boolean input work.

#### Developer

- [x] All new ENV is documented in README
- [x] All new ENV has been added to Heroku Pipeline, Staging and Prod
- [x] ANDI or Wave has been run in accordance to
      [our guide](https://mitlibraries.github.io/guides/basics/a11y.html) and
      all issues introduced by these changes have been resolved or opened as new
      issues (link to those issues in the Pull Request details above)
- [ ] Stakeholder approval has been confirmed (or is not needed)

#### Code Reviewer

- [x] The commit message is clear and follows our guidelines
      (not just this pull request message)
- [x] There are appropriate tests covering any new functionality
- [x] The documentation has been updated or is unnecessary
- [x] The changes have been verified
- [x] New dependencies are appropriate or there were no changes

#### Requires database migrations?

NO

#### Includes new or updated dependencies?

NO

### CODE REVIEW

**Review by matt-bernhardt — COMMENTED** [review]
I'd like to understand what happens (and whether anything needs to happen) to the `transfer_certified` attribute on the Transfer form, and I think I'd like to see a follow-up ticket for the "who you are" partial (although I'm open to pushback there).

The question about spacing on the thesis form is just to make sure I'm not missing anything - I'm not going to demand that it change (the indenting doesn't appear to be "correct" to begin with)

File: app/views/thesis/new.html.erb [hunk 1]
```diff
@@ -103,10 +103,10 @@
...
 
           <%= f.input :graduation_month, label: 'Month', as: :select,
-                               collection: Thesis::VALID_MONTHS,
-                               wrapper_html: { class: 'field-wrap' },
-                               input_html: { class: 'field field-select', data: { msg: Thesis::VALIDATION_MSGS[:graduation_month] },
-                               'aria-describedby': 'thesis_date_ids-hint' } %>
+            collection: Thesis::VALID_MONTHS,
+            wrapper_html: { class: 'field-wrap' },
+               input_html: { class: 'field field-select', data: { msg: Thesis::VALIDATION_MSGS[:graduation_month] },
+               'aria-describedby': 'thesis_date_ids-hint' } %>
```
[the remark this answers is not in the document]
Comment (jazairi) [reply app/views/thesis/new.html.erb:109]: Ah, sorry about that. I had inadvertently changed some of the spacing, and I tried to revert them pre-review, but I seem to have missed this one. Yes, there's no substantive change here, and I will plan to revert this change before merging for our future collective sanity.

File: app/views/transfer/new.html.erb [hunk 2]
```diff
@@ -4,6 +4,46 @@
   <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery-validate/1.17.0/jquery.validate.min.js"></script>
 <% end %>
 
+<h3 class="title title-page">Transfer theses</h3>
+<p class="copy-sup">
+  You are logged in and submitting as 
+  <%= "#{current_user.display_name} (#{current_user.email})" %>. Not you? 
+  <%= link_to "Sign out", destroy_user_session_path, method: :delete %>.
+</p>
```
[the remark this answers is not in the document]
Comment (jazairi) [reply app/views/transfer/new.html.erb:12]: I think that should be pretty straightforward. Let me give it a shot, and if it becomes complicated, I'll create a ticket for it.

File: app/models/transfer.rb [hunk 3]
```diff
@@ -16,6 +16,7 @@ class Transfer < ApplicationRecord
 
   has_many_attached :files
 
+  attr_accessor :transfer_certified
```
[the remark this answers is not in the document]
Comment (jazairi) [reply app/models/transfer.rb:19]: These are good questions. Yes, it's not stored anywhere in the database and is only meant to be a required element in the form, but I'm making a significant assumption that this is the desired behavior. I'll check with Mikki in Slack to see if she wants to add this to the data model.

[the remark this answers is not in the document]
Comment (jazairi) [reply app/models/transfer.rb:19]: Mikki just confirmed that this data shouldn't be stored in the database.

File: app/views/thesis/new.html.erb (hunk 1 above)
[the remark this answers is not in the document]
Comment (jazairi) [reply app/views/thesis/new.html.erb:109]: Never mind. Turns out this _is_ the reverted spacing, so apparently I've just broken the diff forever. I'm not going to attempt to fiddle with it any further.

[the remark this answers is not in the document]
Comment (matt-bernhardt) [reply app/views/thesis/new.html.erb:109]: Yeah, no worries then - someone else will touch this form at some point, and they can fix it if they choose.

File: app/models/transfer.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (matt-bernhardt) [reply app/models/transfer.rb:19]: Ok, cool - this is fine then. I'm not sure it needs a unit test then, either.

File: app/views/transfer/new.html.erb (hunk 2 above)
[the remark this answers is not in the document]
Comment (jazairi) [reply app/views/transfer/new.html.erb]: @matt-bernhardt I just pushed a commit to use the `you_are` partial. If this approach looks good, I can update the render link in `thesis/new` once I rebase.

[the remark this answers is not in the document]
Comment (matt-bernhardt) [reply app/views/transfer/new.html.erb]: Yeah, this looks good to me. Thanks!

**Review by matt-bernhardt — APPROVED** [review]
The new changes look good - thanks! :shipit:

**Review by matt-bernhardt — APPROVED** [review]
This still looks fine - well done :-)

:shipit:

**Pull request merged**